cEP-0029.md

```diff
@@ -0,0 +1,348 @@
+              ('use_spaces', <Setting object(key='use_spaces', value='True', origin='/Users/userx/Documents/coala/coala/.coafile', from_cli=False, to_append=False) at 0x1092cd898>)]), defaults=None,
+              language=None, name='Default') at 0x108bdf550>)])
+```
+
+into
+
+```python
+OrderedDict([('Default',
+              OrderedDict([('files',
+                            ['*.py',
+                             'coalib/**/*.py',
+                             './coala',
+                             'tests/**/*.py',
+                             'docs/conf.py']),
+                           ('ignore',
+                            ['tests/bearlib/languages/documentation/documentation_extraction_testdata/*.py',
+                             'tests/collecting/collectors_test_dir/bears/incorrect_bear.py',
+                             'coalib/misc/Asyncio.py']),
+                           ('max_line_length', 80),
+                           ('use_spaces', True)]))])
+```
+into
+
+
+### Tool to convert .coafile to TOML config file
+
+A new flag will be created to allow .coafile to
+be converted to TOML config files and vice versa.
+
+```bash
+$ coala --create-toml .coafile
+$ coala --create-coafile .coafile.toml
+```
+
+The loaded sections from either .coafile or .coafile.toml
+can be supplied to:
+- TomlConfWriter to create TOML config file
+- ConfWriter to create coafile
+
+
+### Support coala-quickstart to generate config files in TOML.
+
+coala-quickstart will be made to generate config files in TOML.
+A new option -T that will allow config files to be created in
+TOML.
+
+New functions `write_toml_config` and `write_toml_info` will
+be created which will perform the functions similar to existing
+`write_coafile` and `write_info` functions but with respect to TOML.
+
+Regarding green mode, I will create `write_toml_sections` function
+similar to `write_sections` for TOML. Also the function
+`generate_green_mode_sections` will be modified to accommodate
+for TOML config file.
+
+
+### Develop TOMLLintBear
+
+coala-bears currently has bear for TOML called `TOMLBear.py` which
+uses a external linter. But it requires Go to be installed.
+So this bear can mainly be used only by Go users.
+
+The task is to create `TOMLLintBear` in python. This way it can
+be used by all users without any additional language requirements
+as python comes installed in most computers.
+
+The bear shall catch syntax errors in a TOML file. This shall be
+done in 2 passes.
+- In the first pass, the TOML file shall be validated per character
+- In the second pass, the TOML file shall be validated per line
+
+This approach is similar to the toml library. As part of this project
+the bear shall be a validator and only check the validity of the TOML
+file. In the future when style guidelines are created formatting
+capability can be added.
+
+
+### A repository of good config files to allow users to easily bootstrap their projects.
+
```

Comment:
I feel that this project is related but some example config files and some community-provided examples should be there from the start. Also, don't you think that this should be included within the TOML support itself, not as a separate project?

Reply:
@margobra8 I will provide example config files and documentation in TOML from the start to help users write configuration files in TOML like http://docs.coala.io/en/latest/Users/coafile.html.

`A repository of good config files to allow users to easily bootstrap their projects.` is about creating a separate repo similar to https://github.com/github/gitignore and https://github.com/moremoban/pypi-mobans to allow easy bootstrapping of projects.

Reply:
@PrajwalM2212 the thing I'm trying to get to is if you actually consider generating another repo with templates as part of the initial requirements of the project or as future development or to be implemented later.

Reply:
As part of GSoC, another repo will be created with few templates as part of the initial requirements of the project.  A minimal version of `A repository of good config files` will be created during GSoC.